web/profiles/custom/yalesites_profile/modules/custom/ys_core/src/CoreSettingsManager.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Drupal\ys_core;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Service for managing YaleSites core settings.
+ */
+class CoreSettingsManager {
+
+  /**
+   * Configuration Factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $yaleCoreSettings;
+
+  /**
+   * Construct function for core settings manager.
+   *
+   * @param Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Configuration factory.
+   */
+  public function __construct(ConfigFactoryInterface $configFactory) {
+    $this->yaleCoreSettings = $configFactory->getEditable('ys_core.site');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+    );
+  }
+
+  /**
+   * Gets core setting from config.
+   *
+   * @param string $setting_name
+   *   Setting machine name.
+   */
+  public function getSetting($setting_name) {
```

Comment:
Please include @retrun type params (I believe it is an immutable config) and also type hint at least the arguments on the function declaration. 
`public function getSetting(string $setting_name)`